lib/cucumber/factory.rb

```diff
@@ -14,16 +15,16 @@ class Factory
 
     NAMED_CREATION_STEP_DESCRIPTOR = {
       :kind => :Given,
-      :pattern => /^"([^\"]*)" is an? (.+?)( \(.+?\))?#{ATTRIBUTES_PATTERN}?$/,
+      :pattern => /^"([^\"]*)" is an? (.+?)( \(.+?\))?#{ATTRIBUTES_PATTERN}?$/m,
       # we cannot use vararg blocks here in Ruby 1.8, as explained by Aslak: http://www.ruby-forum.com/topic/182927
-      :block => lambda { |a1, a2, a3, a4, a5| Cucumber::Factory.send(:parse_named_creation, self, a1, a2, a3, a4, a5) }
+      :block => lambda { |a1, a2, a3, a4, a5, a6| Cucumber::Factory.send(:parse_named_creation, self, a1, a2, a3, a4, a5, a6) }
     }
 
     CREATION_STEP_DESCRIPTOR = {
       :kind => :Given,
-      :pattern => /^there is an? (.+?)( \(.+?\))?#{ATTRIBUTES_PATTERN}$/,
+      :pattern => /^there is an? (.+?)( \(.+?\))?#{ATTRIBUTES_PATTERN}$/m,
```

Comment:
I think there has been a misunderstanding on how multiline arguments in Cucumber work.

What cucumber expects of you is to define a step like this:
```
Given(^/the following multiline text:$/) do |text|
  # text is just a regular string here
end
```
i.e. you just expect an additional argument and Cucumber will parse the multiline string and pass it to you.

Similarly for multiline tables
```
Given(/^the following table:$/) do |table|
  # table is an instance of Cucumber::MultilineArgument::DataTable
end
```
See [cucumber rdoc for DataTable](http://www.rubydoc.info/gems/cucumber/Cucumber/MultilineArgument/DataTable), I think you want the `#rows_hash` method.

That means you should not capture the multiline string / table yourself, and not need a `/m` flag for the regexp.

Have you tried your changes with an actual cucumber run? I think your code probably only works inside the `cucumber_factory` specs?